### ISSUE TITLE
Admin user and #admins: room

### ISSUE THREAD

**Issue by Thatoo** [YunoHost-Apps/conduit_ynh#13]
For what I understood from https://gitlab.com/famedly/conduit/-/issues/12, at install, a  #admins:server.name room should be created and the first user that register to conduit join automatically this room.

During [my first test with conduit](https://github.com/YunoHost-Apps/conduit_ynh/issues/12), I could install, register a user and login but with this user, I could not join #admins:server.name and no even verify if #admins:server.name room exists.

Maybe we should ask to select an admin user at install and be sure this user join the room at install.

Of course this issue is also related to the lack of OIDC and LDAP support. I'm just writing it down to track the issue.

**Comment by Thatoo** [comment]
Well I tried to install a second time and this time the room was created automatically.
But then I install a third time and the room was not created. Weird.

**Comment by Thatoo** [comment]
So apparently, it would be because robots would actually create a user faster and get the room admin.

**Comment by Thatoo** [comment]
So the solution would be to ask to choose a user to be the admin of the server and its password and create this user (as ldap isn't working with conduit) at install. Something similar as libreerp is doing.
I need to find out how to create a user at install time with command line.

**Comment by Thatoo** [comment]
Something sounds weird to me.
At the same time, in manifest.toml it is written that If registration is yes, to protect your server and the federation from spammer, federation will be deactived by default.
However, in ths install script, it is coded like
`federation="true"`
so value of federation don't seem to depend on value of registration, does it?

**Comment by Thatoo** [comment]
Actually, to make it easy, I would not ask about registration at install time.
I would deactivate it anyway and deactivate federation and activate only 
`registration_token="password"`
and explain that the first user should be created thanks to this token because it becomes the admin user who has access to the admin room and can manage the server.
Then, only in the config panel, I would let option to activate registration without token and let activate federation.
I'll make a PR in this sense.

**Comment by Thatoo** [comment]
Well, I face a weird situation.
I set up conduit.toml like that

```
allow_registration = false
registration_token = "STORNG_PASSWORD"
allow_federation = false
```

and thus, if I was not fast enough, I would not become admin, an other user would become admin even though the TOKEN was pretty strong.

Then, I realized that if I open a new private window in Firefox and I got directly to https://mydomain/element/#/home , then it would log me as a visitor.

So I discovered and reported upstream that a visitor can become admin if no user register fast enough.

So it would be better to limit conduit to all_user only (not visitors) at install but switch to visitors when federation is activated.

**Comment by Thatoo** [comment]
Here is the upstream issue : https://gitlab.com/famedly/conduit/-/issues/415

We need that to be solved upstream to solve this issue.

**Comment by Thatoo** [comment]
It will be solved with the next conduit version : https://gitlab.com/famedly/conduit/-/merge_requests/591
Then we will be able to close this issue.

**Comment by ericgaspar** [comment]
Is the issue fixed with 0.7.0 release?

**Comment by tituspijean** [comment]
Assuming it has been solved, closing.
